rskj-core/src/main/java/co/rsk/net/sync/PeersInformation.java

```diff
@@ -44,26 +44,31 @@ public SyncPeerStatus getOrRegisterPeer(MessageChannel messageChannel) {
     }
 
     public SyncPeerStatus getPeer(NodeID nodeID) {
-        // TODO(mc) check expiration
         return this.peerStatuses.get(nodeID);
     }
 
     public Optional<MessageChannel> getBestPeer() {
-        return peerStatuses.entrySet().stream()
-                .filter(e -> !e.getValue().isExpired(syncConfiguration.getExpirationTimePeerStatus()))
-                .max(this::bestPeerComparator)
+        return getCandidates()
                 .map(Map.Entry::getValue)
+                .max(SyncPeerStatus::peerTotalDifficultyComparator)
                 .map(SyncPeerStatus::getMessageChannel);
     }
 
-    public Set<NodeID> knownNodeIds() {
-        return peerStatuses.keySet();
+    private Stream<Map.Entry<NodeID,SyncPeerStatus>> getCandidates(){
+        return peerStatuses.entrySet().stream()
+                .filter(e -> !e.getValue().isExpired(syncConfiguration.getExpirationTimePeerStatus()))
+                .filter(e -> syncInformation.hasGoodReputation(e.getKey()))
```

Comment:
Are you leaving out the ones with good reputation? :scream:

Reply:
I'm leaving the ones with good reputation in and the expireds out